vpr/src/place/simpleRL_move_generator.h

```diff
@@ -211,10 +212,12 @@ class SimpleRLMoveGenerator : public MoveGenerator {
      *
      *   @param agent std::unique_ptr to the agent. Only EpsilonGreedyAgent and SoftmaxAgent types are accepted
      *   by the constructor. If other types are passed, a compile error would be thrown.
+     *
+     *   @param is_multi_layer A boolean value to indicate whether the placement is multi-layer or not
      */
     template<class T,
              class = typename std::enable_if<std::is_same<T, EpsilonGreedyAgent>::value || std::is_same<T, SoftmaxAgent>::value>::type>
-    explicit SimpleRLMoveGenerator(std::unique_ptr<T>& agent);
+    explicit SimpleRLMoveGenerator(std::unique_ptr<T>& agent, bool is_multi_layer);
```

Comment:
Do you need to pass is_multi_layer around, or could you store it in the architecture or placement context?